api/src/main/java/org/openmrs/layout/name/NameSupport.java

```diff
@@ -23,18 +23,15 @@
 	
 	static Logger log = LoggerFactory.getLogger(NameSupport.class);
 	
-	public NameSupport() {
-		if (singleton == null) {
-			singleton = this;
-		}
+	private NameSupport() {
+		// hiding default public constructor
 	}
 	
 	public static NameSupport getInstance() {
 		if (singleton == null) {
-			throw new RuntimeException("Not Yet Instantiated");
-		} else {
-			return singleton;
+			singleton = new NameSupport();
```

Comment:
Isn't this resulting into a bug caused by a bean different from this? https://github.com/openmrs/openmrs-module-legacyui/blob/master/omod/src/main/resources/webModuleApplicationContext.xml#L754